cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -272,11 +273,12 @@ public void getStatus(HttpRequest request, HttpResponder responder,
     ApplicationId applicationId = new ApplicationId(namespaceId, appId, versionId);
     if (SCHEDULES.equals(type)) {
       JsonObject json = new JsonObject();
-      ScheduleId scheduleId = applicationId.schedule(programId);
       ApplicationSpecification appSpec = store.getApplication(applicationId);
       if (appSpec == null) {
         throw new NotFoundException(applicationId);
       }
+      // Use the actual version from appSpec
+      ScheduleId scheduleId = applicationId.getAppReference().app(appSpec.getAppVersion()).schedule(programId);
```

Comment:
This is not needed. ScheduleId always set version to be "-SNAPSHOT", you can check the scheduleId constructor to see this.

Reply:
got it, will revert it